app/models/article.rb

```diff
@@ -45,6 +45,7 @@ class Article < ApplicationRecord
   validate :validate_collection_permission
   validate :validate_liquid_tag_permissions
   validate :past_or_present_date
+  validate :canonical_url_must_not_has_spaces
```

Comment:
this should be `canonical_url_must_not_have_spaces` as the error string

Reply:
@rhymes I’ll change it! what about the error message?